src/test/java/com/iota/iri/controllers/TipsViewModelTest.java

```diff
@@ -136,4 +157,67 @@ public void tipsEmtpyAfterClear() throws Exception {
         assertEquals(0, tipsVM.size());
     } 
 
-}
\ No newline at end of file
+    @Test
+    public void solidTipsPopulatedWhenEmpty() throws Exception {
+        final int tipCount = 10;
+        Set<Hash> tips = new HashSet<>();
+
+        for (int i = 0; i < tipCount; i++) {
+            TransactionViewModel newTip = new TransactionViewModel(
+                getRandomTransactionWithTrunkAndBranch(Hash.NULL_HASH, Hash.NULL_HASH), getRandomTransactionHash());
+            newTip.updateSolid(true);
+
+            newTip.store(tangle);
+            tips.add(newTip.getHash());
+        }
+
+        TipsViewModel tipsVM = new TipsViewModel(tangle);
+
+        Hash tip = tipsVM.getRandomSolidTipHash();
+
+        assertNotNull(tip);
+        assertTrue(tips.contains(tip));
+    }
+
+    @Test
+    public void nonSolidTipsAreIgnoredWhenPopulatingSolidTips() throws Exception {
+        TransactionViewModel solidTip = new TransactionViewModel(
+            getRandomTransactionWithTrunkAndBranch(Hash.NULL_HASH, Hash.NULL_HASH), getRandomTransactionHash());
+        TransactionViewModel nonSolidTip = new TransactionViewModel(
+            getRandomTransactionWithTrunkAndBranch(Hash.NULL_HASH, Hash.NULL_HASH), getRandomTransactionHash());
+
+        solidTip.updateSolid(true);
+
+        solidTip.store(tangle);
+        nonSolidTip.store(tangle);
+
+        TipsViewModel tipsVM = new TipsViewModel(tangle);
+
+        Hash tip = tipsVM.getRandomSolidTipHash();
+
+        assertEquals(solidTip.getHash(), tip);
+    }
+
+    @Test
+    public void getLatestSolidTipsHappyFlow() throws Exception {
+        final int total = 50;
+        final int count = 10;
+
+        TipsViewModel tipsVM = new TipsViewModel(tangle);
+
+        for (int i = 0; i < total; i++) {
+            TransactionViewModel newTip = new TransactionViewModel(
+                getRandomTransactionWithTrunkAndBranch(Hash.NULL_HASH, Hash.NULL_HASH), getRandomTransactionHash());
+            
+            tipsVM.addTipHash(newTip.getHash());
+            tipsVM.setSolid(newTip.getHash());
+        }
+
+
+        List<Hash> tips = tipsVM.getLatestSolidTips(count);
+
+        assertNotNull(tips);
+        assertEquals(count, tips.size());
```

Comment:
maybe add a check for maintaining order? i.e. that these are the *last* 10 tips?